### PR TITLE
Business owner test cases.

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public void startUp() throws Exception {
+        appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER, TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+    }
+
+    @Test(dataProvider = "validUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testAdminDashboardLoginWithValidUsers(String userName, String password) throws Exception {
+        APPMAdminDashboardRestClient appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        HttpResponse response = appmAdminDashboardRestClient.login(userName, password);
+        JSONObject publisherLogoutJsonObject = new JSONObject(response.getData());
```
Comment (jenananthan) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: publisherLogoutJsonObject - > loginResponse ??

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        String dataResponse = publisherLogoutJsonObject.getString(AppmTestConstants.ERROR);
+        assertTrue(dataResponse == "false", "Login to Admin dashboard is not allowed for user : " + userName + " who " +
```
Comment (jenananthan) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: should use .equals() to compare the values for two string objects 
http://stackoverflow.com/questions/7520432/what-is-the-difference-between-vs-equals-in-java

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java (hunk 1 above)
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java (hunk 2 above)
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        appmAdminDashboardRestClient.logout();
+    }
+
+    @Test(dataProvider = "inValidUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testAdminDashboardLoginWithInValidUsers(String userName, String password) throws Exception {
+        APPMAdminDashboardRestClient appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        HttpResponse response = appmAdminDashboardRestClient.login(userName, password);
+        JSONObject publisherLogoutJsonObject = new JSONObject(response.getData());
+        String dataResponse = publisherLogoutJsonObject.getString(AppmTestConstants.ERROR);
+        assertTrue(dataResponse == "true", "Login to Admin dashboard is allowed for user : " + userName + " who " +
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        User adminUser = appMServer.getSuperTenant().getTenantUser(AppmTestConstants.TestUsers.ADMIN);
+        return new Object[][]{
+                new Object[]{adminUser.getUserName(), adminUser.getPassword()}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] inValidUserModeDataProvider() throws Exception {
+        User appCreator = appMServer.getSuperTenant().getTenantUser("AppCreator");
+        User appPublisher = appMServer.getSuperTenant().getTenantUser("AppPublisher");
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package org.wso2.appmanager.integration.test.cases.admin.dashboard.authentication;
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        userName = adminUser.getUserName();
+        String password = adminUser.getPassword();
+        appmAdminDashboardRestClient.login(userName, password);
+    }
+
+    @Test(description = TEST_DESCRIPTION)
+    public void testAdminDashboardLogin() throws Exception {
+        HttpResponse logoutResponseData = appmAdminDashboardRestClient.logout();
+        JSONObject publisherLogoutJsonObject = new JSONObject(logoutResponseData.getData());
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        String dataResponse = publisherLogoutJsonObject.getString(AppmTestConstants.ERROR);
+        assertTrue(dataResponse == "false", "Logout from Admin dashboard is not allowed for user : " + userName +
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogout.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/AddNewBusinessOwnerTestCase.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * Test case which verifies the ability of admin user of adding a new business owner.
+ */
+public class AddNewBusinessOwnerTestCase {
+    private static final String TEST_DESCRIPTION = "Verify adding a new business owner";
+    private static AutomationContext appMServerSuperTenant;
+    private APPMAdminDashboardRestClient appmAdminDashboardRestClient;
+    private String businessOwnerName = "AddNewBusinessOwnerTestCase_";
+
+
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/AddNewBusinessOwnerTestCase.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/AddNewBusinessOwnerTestCase.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    public void startUp() throws Exception {
+        appMServerSuperTenant = new AutomationContext(AppmTestConstants.APP_MANAGER, TestUserMode.SUPER_TENANT_ADMIN);
+    }
+
+    @Test(dataProvider = "validUserModeDataProvider",description = TEST_DESCRIPTION)
+    public void testAddNewBusinessOwnerWithValidUsers(String userName, String password, AutomationContext
+            automationContext) throws Exception {
+        String backEndUrl = automationContext.getContextUrls().getWebAppURLHttps();
+        appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        // login to admin dashboard rest client.
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/AddNewBusinessOwnerTestCase.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/DeleteBusinessOwnerTestCase.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    @Test(dataProvider = "validUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testDeleteBusinessOwnerWithValidUsers(String userName, String password, AutomationContext
+            automationContext) throws Exception {
+        String backEndUrl = automationContext.getContextUrls().getWebAppURLHttps();
+        appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        appmAdminDashboardRestClient.login(userName, password);
+
+        int businessOwnerId = createBusinessOwner(appmAdminDashboardRestClient, userName);
+        HttpResponse deleteBusinessOwnerResponse =  appmAdminDashboardRestClient.deleteBusinessOwner(businessOwnerId);
+        // logout from admin dashboard rest client.
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/DeleteBusinessOwnerTestCase.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/UpdateBusinessOwnerTestCase.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * Test case which verifies the ability of admin user of updating a given business owner.
+ */
+public class UpdateBusinessOwnerTestCase {
+    private static final String TEST_DESCRIPTION = "Verify retrieving given business owner.";
+    private APPMAdminDashboardRestClient appmAdminDashboardRestClient;
+    private String businessOwnerName = "UpdateBusinessOwnerTestCase";
+    private static AutomationContext appMServerSuperTenant;
+
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/businessowner/UpdateBusinessOwnerTestCase.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/APPMAdminDashboardRestClient.java [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class APPMAdminDashboardRestClient {
+    private String backEndUrl;
+    private Map<String, String> requestHeaders = new HashMap<String, String>();
+
+    public APPMAdminDashboardRestClient(String backEndUrl) throws MalformedURLException {
+        this.backEndUrl = backEndUrl;
+        if (requestHeaders.get(AppmTestConstants.CONTENT_TYPE) == null) {
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/APPMAdminDashboardRestClient.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/AppmTestConstants.java [hunk 13]
```diff
@@ -22,6 +22,16 @@
 
     public static final String AVERAGE = "average";
 
+    public static final String BUSINESS_OWNER_ID = "businessOwnerId";
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/AppmTestConstants.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

File: modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/bean/BusinessOwnerCreateRequest.java [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+        List<BusinessOwnerProperty> businessOwnerPropertiesList = businessOwner.getBusinessOwnerPropertiesList();
+        JSONObject businessOwnerCustomProperties = new JSONObject();
+        if (businessOwnerPropertiesList != null) {
+            for(int i = 0; i < businessOwnerPropertiesList.size(); i++) {
+                JSONArray propertiesArray = new JSONArray();
+                try {
+                    propertiesArray.put(0, businessOwnerPropertiesList.get(i).getPropertyValue());
+                    propertiesArray.put(1, businessOwnerPropertiesList.get(i).isShowingInStore());
+                    businessOwnerCustomProperties.put(businessOwnerPropertiesList.get(i).getPropertyId(),
```
Comment (lakshani) [inline modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/bean/BusinessOwnerCreateRequest.java]: Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea

**Pull request merged**